### PR TITLE
Issue 5687 - UI - sensitive information disclosure

### PR DESCRIPTION
**Pull request by mreynolds389**
Bug Description: 

In several places either the clear text password or the password hash can be read by unprivileged users.

Fix Description: 

When processing user entries do not attempt to decode userpassword.  When setting the password for chaining or replication agreements/repl manager write the password to a temporary file that can be passed to the CLI.

Also, improved user add wizard allowing to search attributes

relates: https://github.com/389ds/389-ds-base/issues/5687

### CODE REVIEW

File: src/cockpit/389-console/src/lib/replication/replAgmts.jsx [hunk 1]
```diff
@@ -1193,31 +1195,122 @@ export class ReplAgmts extends React.Component {
...
+                                                        });
+                                                        cockpit.spawn(rm_cmd, { superuser: true }); // Remove files
+                                                    })
+                                                    .done(_ => {
+                                                        //
+                                                        // Call the full cmd
+                                                        //
+                                                        log_cmd('saveAgmt', 'update replication agreement', cmd);
+                                                        cockpit
+                                                                .spawn(full_cmd, {superuser: true, err: "message"})
```
Comment (droideck) [inline src/cockpit/389-console/src/lib/replication/replAgmts.jsx]: When I try to create an agreement this error is displayed:

```
Uncaught ReferenceError: full_cmd is not defined
    at Function.<anonymous> (index.js:2:2007712)
    at cockpit.js:1:12867
    at cockpit.js:1:12978
    at L (cockpit.js:1:12383)
```

File: src/cockpit/389-console/src/lib/replication/replAgmts.jsx [hunk 2]
```diff
@@ -1193,31 +1195,122 @@ export class ReplAgmts extends React.Component {
...
-        log_cmd('saveAgmt', 'update replication agreement', cmd);
+
+        const passwdFile = "/tmp/ds-passwd-file-" + new Date().getTime() + ".txt";
+        const bsPasswdFile = "/tmp/ds-bs-passwd-file-" + new Date().getTime() + ".txt";
+        const rm_cmd = ["rm", passwdFile, ];
+        const create_file_cmd = ["touch", passwdFile, bsPasswdFile];
+        const chmod_cmd = ["chmod", "600", passwdFile, bsPasswdFile ];
+        const update_cmd = [
+            '/bin/sh', '-c',
+            '/usr/bin/echo -e \'' + passwd + '\' >> ' + passwdFile
```
Comment (droideck) [inline src/cockpit/389-console/src/lib/replication/replAgmts.jsx]: I wonder if the command will still appear on `ps -ef` output...

I think we can use something from cockpit-spawn options:
https://cockpit-project.org/guide/latest/cockpit-spawn

Maybe something like this with .stream()?
https://github.com/389ds/389-ds-base/blob/main/src/cockpit/389-console/src/lib/monitor/replMonitor.jsx#L1116-L1177

**Review by droideck — CHANGES_REQUESTED** [review]

File: src/cockpit/389-console/src/lib/replication/replAgmts.jsx (hunk 2 above)
Comment (droideck) [reply src/cockpit/389-console/src/lib/replication/replAgmts.jsx]: We can ask for the password with something like that also:
`read -s -p "Password: " password ; echo $password > passwdFile`

Or implement the "user input" functionality into our CLI.

Comment (vashirov) [reply src/cockpit/389-console/src/lib/replication/replAgmts.jsx]: Yes, it's still visible:
```
2023/03/02 03:16:40 CMD: UID=0     PID=3438   | /bin/sh -c /usr/bin/echo -e 'test' >> /tmp/ds-passwd-file-1677745001625.txt
2023/03/02 03:16:40 CMD: UID=0     PID=3439   | /bin/sh -c /usr/bin/echo -e 'test' >> /tmp/ds-passwd-file-1677745001625.txt
```

Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/replication/replAgmts.jsx]: Ugh...  This is also how we've been calling dscreate in the UI for years.  Gonna have to make more changes to the CLI to support password prompting...

File: src/cockpit/389-console/src/lib/replication/replSuffix.jsx [hunk 3]
```diff
@@ -231,24 +232,23 @@ export class ReplSuffix extends React.Component {
...
+
+
+        // Something changed, perform the update
+        const config = {
+            cmd: cmd,
+            promptArg: "--bind-passwd-prompt",
+            passwd: passwd,
+            addNotification: this.props.addNotification,
+            success_msg: `Successfully enabled replication for "${this.props.suffix}"`,
+            error_msg: `Failed to enable replication for "${this.props.suffix}"`,
```
Comment (droideck) [inline src/cockpit/389-console/src/lib/replication/replSuffix.jsx:244]: I'm not sure why but enable replication fails for me with the next error:

```
Danger alert:
Failed to enable replication for "dc=example,dc=com": usage: dsconf [-h] [-v] [-D BINDDN] [-w BINDPW] [-W] [-y PWDFILE] [-b BASEDN]
              [-Z] [-j]
              instance
              {backend,backup,chaining,config,directory_manager,monitor,plugin,pwpolicy,localpwp,replication,repl,repl-agmt,repl-winsync-agmt,repl-tasks,sasl,security,schema,repl-conflict}
              ...
dsconf: error: unrecognized arguments: undefined
```

The steps to reproduce:
[Screencast from 03-03-2023 08:11:10 AM.webm](https://user-images.githubusercontent.com/5634892/222770561-4fc21648-1f45-4a73-81ce-d391f4bd025f.webm)

**Review by droideck — CHANGES_REQUESTED** [review]

File: src/cockpit/389-console/src/lib/replication/replSuffix.jsx (hunk 3 above)
Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/replication/replSuffix.jsx:244]: Fixing this now...

Comment (mreynolds389) [reply src/cockpit/389-console/src/lib/replication/replSuffix.jsx:244]: @droideck - should be fixed now...

**Review by droideck — APPROVED** [review]
Looks great!
I'm totally happy with it.
And I'd give some time for @vashirov to check if he wants to, as it's past his working time now. :)

**Review by vashirov — APPROVED** [review]
LGTM too, thank you!

**Pull request merged**